### PR TITLE
Update release docs

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This PR simply adds some minor updates after the latest release process. Note that it slightly changed since the 2.3.2 version was released.

### Benefits
 
Updated docs

### Possible drawbacks

N/A

### Applicable issues

N/A

### Additional information

N/A

### CODE REVIEW

File: docs/developer/release-process.md [hunk 1]
```diff
@@ -97,49 +97,47 @@ helm dependency update ./chart/kubeapps
...
+
+Check the outdated [golang dependencies](../../go.mod) by running the following (from [How to upgrade and downgrade dependencies](https://github.com/golang/go/wiki/Modules#how-to-upgrade-and-downgrade-dependencies)):
 
 ```bash
 go mod tidy
 go list -u -f '{{if (and (not (or .Main .Indirect)) .Update)}}{{.Path}}: {{.Version}} -> {{.Update.Version}}{{end}}' -m all 2> /dev/null
-go-mod-upgrade
-# go get -u -t ./... # upgrades to minor or patch releases
-# go get -u=patch  -t ./... # upgrades just to patch releases
+go get -u ./...
```
Comment (absoludity) [inline docs/developer/release-process.md:116]: Deja vu, but doesn't the above already try to "upgrade to the latest version for all direct and indirect dependencies of the current module", in which case, the following sentence doesn't make sense. I think we should either be recommending doing `go get -u ./...` and dealing with the fallout manually (won't this try to upgrade dependencies that we can't yet upgrade? Fails for me locally with `module declares its path as: github.com/moby/spdystream but was required as: github.com/docker/spdystream`, but that could be transitional), or recommending doing `go-mod-upgrade` and selecting the packages we want to upgrade and leaving those that we can't?

File: docs/developer/release-process.md [hunk 2]
```diff
@@ -97,49 +97,47 @@ helm dependency update ./chart/kubeapps
...
-
-Since the chart that we host in the Kubeapps repository is only intended for development purposes, we need to synchronize it with the official one in the [bitnami/charts repository](https://github.com/bitnami/charts/tree/master/bitnami/kubeapps).
+> As part of this release process, the dashboard deps _must_ be updated, the golang deps _should_ be updated, the rust deps _should_ be updated and the security check _must_ be performed.
 
-To this end, our CI system will automatically (in the `sync_chart_from_bitnami` workflow, as described in the [CI documentation](./ci.md).) send a PR with the current development changes to [their repository](https://github.com/bitnami/charts/pulls) whenever a new release is triggered.
-Once the PR has been created, wait for someone from the Bitnami team to review and accept it.
+#### Send a PR with the upgrades
 
-> If the PR solely includes minor changes in the image versions, this wait can be safely skipped.
+Now create a Pull Request containing all these changes (only if only minor and patch have been bumped up) and wait until for another Kubeapps maintainer to review and accept it.
```
Comment (absoludity) [inline docs/developer/release-process.md:138]:
```suggestion
Now create a Pull Request containing all these changes (only if only minor and patch have been bumped up) and wait until for another Kubeapps maintainer to review and accept so you can merge it.
```
Also, the "only if only minor...." is a bit confusing, I'm not sure if the second only is intentional, in which case, perhaps "only if no major versions have been bumped up", but I think you mean rather that "(there should only ever be minor and patch versions updated)"?

File: docs/developer/release-process.md [hunk 3]
```diff
@@ -165,21 +163,39 @@ For doing so, execute the following commands:
...
+> ```
 
-> When a new tag is detected, Bitnami will automatically build a set of container images based on the tagged commit. They later will be published in [the Dockerhub image registry](https://hub.docker.com/search?q=bitnami%2Fkubeapps&type=image).
+A new tag pushed to the repository will trigger, apart from the usual test and build steps, a _release_ [workflow](https://circleci.com/gh/kubeapps/workflows) as described in the [CI documentation](./ci.md). An example of the triggered workflow is depicted below:
 
-## 4 - Complete the GitHub release notes
+![CircleCI workflow after pushing a new tag](../img/ci-workflow-release.png "CircleCI workflow after pushing a new tag")
+
+> When a new tag is detected, Bitnami will automatically build a set of container images based on the tagged commit. They later will be published in [the Bitnami Dockerhub image registry](https://hub.docker.com/search?q=bitnami%2Fkubeapps&type=image).
+> Please note that this process is independent from the Kubeapps release process.
```
Comment (absoludity) [inline docs/developer/release-process.md:180]: It's not independent (we're triggering it), but it is run outside the control of the Kubeapps release process - is that what you mean?

File: docs/developer/release-process.md [hunk 4]
```diff
@@ -165,21 +163,39 @@ For doing so, execute the following commands:
...
 
+## 4 - Manually review the PR created in the bitnami/charts repository
+
+Since the chart that we host in the Kubeapps repository is only intended for development purposes, we need to synchronize it with the official one in the [bitnami/charts repository](https://github.com/bitnami/charts/tree/master/bitnami/kubeapps).
+
+To this end, our CI system will automatically (in the `sync_chart_to_bitnami` workflow, as described in the [CI documentation](./ci.md).) send a PR with the current development changes to [their repository](https://github.com/bitnami/charts/pulls) whenever a new release is triggered.
+Once the PR has been created, have a look at it (eg. remove any development changes that should not be released) and wait for someone from the Bitnami team to review and accept it.
+
+> Some issues can arise here, so please check the app versions are being properly updated at once and ensure you have the latest changes in the PR branch. Note that the [bitnami-bot](https://github.com/bitnami-bot) usually performs some automated commits to the main branch that might collide with the changes in our PR.
+
```
Comment (absoludity) [inline docs/developer/release-process.md:198]: In particular, currently it'll release a new version of the chart with the updated images? (see #3143 )

**Review by absoludity — APPROVED** [review]
A couple of thoughts, but +1

File: docs/developer/release-process.md (hunk 1 above)
Comment (antgamdia) [reply docs/developer/release-process.md:116]: Yes, you're right `go get -u ./...` would perform the upgrade (usually), however, there are some cases (as you faced) in which it is not possible. In that case, the `go-mod-upgrade` could help, as it just updates the ones you select.
I'll try to reword it; I agree currently is quite confusing, sorry :S

**Pull request merged**